common-pages/src/NavBarFilterButton.tsx

```diff
@@ -0,0 +1,136 @@
+//******************************************************************************************************
+//  NavBarFilterButton.tsx - Gbtc
+//
+//  Copyright © 2019, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  08/07/2024 - G. Santos
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+import React from 'react';
+import 'moment';
+import { OpenXDA, SystemCenter } from '@gpa-gemstone/application-typings';
+
+interface S { ID: number }
```

Comment:
would `S = SystemCenter.Types.DetailedMeter|SystemCenter.Types.DetailedAsset|SystemCenter.Types.DetailedLocation| OpenXDA.Types.AssetGroup)` make sense?

Reply:
We could, but then the compiler would complain about S not having ID as a property, requiring a cast in the body of the component.

Reply:
but S would have an ID